rmf_human_detector/src/HumanDetector.cpp

```diff
@@ -0,0 +1,217 @@
+#include <filesystem>
+#include <rclcpp/wait_for_message.hpp>
+#include <sensor_msgs/msg/camera_info.hpp>
+#include <rmf_human_detector/HumanDetector.hpp>
+
+
+HumanDetector::HumanDetector()
+: Node("human_detector"), _data(std::make_shared<Data>())
+{
+  make_detector();
+
+  _data->_obstacles_pub = this->create_publisher<Obstacles>(
+    "/rmf_obstacles",
+    rclcpp::SensorDataQoS()
+  );
+
+  _data->_image_detections_pub =
+    this->create_publisher<sensor_msgs::msg::Image>(
+    _data->_image_detections_topic,
+    rclcpp::QoS(10).reliable()
+    );
+
+  _data->_image_sub = this->create_subscription<sensor_msgs::msg::Image>(
+    _data->_camera_image_topic,
+    rclcpp::SensorDataQoS(),
+    [=](const sensor_msgs::msg::Image::ConstSharedPtr& msg)
+    {
+      // perform detections
+      auto [rmf_obstacles, image_detections] = _data->_detector->image_cb(msg);
+
+      // populate fields like time stamp, etc
+      for (auto& obstacle : rmf_obstacles.obstacles)
+      {
+        obstacle.header.stamp = this->get_clock()->now();
+      }
+
+      // publish rmf_obstacles & image with bounding boxes
+      _data->_obstacles_pub->publish(rmf_obstacles);
+      _data->_image_detections_pub->publish(image_detections);
+    });
+
+  _data->_camera_pose_sub = this->create_subscription<tf2_msgs::msg::TFMessage>(
+    _data->_camera_pose_topic,
+    rclcpp::SensorDataQoS(),
+    [=](const tf2_msgs::msg::TFMessage::ConstSharedPtr& msg)
+    {
+      for (auto& transformStamped : msg->transforms)
+      {
+        if (transformStamped.header.frame_id == _data->_camera_parent_name
+        && transformStamped.child_frame_id ==  _data->_camera_name)
+        {
+          _data->_detector->camera_pose_cb(transformStamped.transform);
+        }
+      }
+    });
+
+  auto qos_profile = rclcpp::QoS(10);
+  qos_profile.transient_local();
+  _data->_building_map_sub =
+    this->create_subscription<rmf_building_map_msgs::msg::BuildingMap>(
+    "/map",
+    qos_profile,
+    [=](const rmf_building_map_msgs::msg::BuildingMap::ConstSharedPtr& msg)
+    {
+      if (msg->levels.empty())
+      {
+        RCLCPP_ERROR(this->get_logger(), "Received empty building map");
+        return;
+      }
+
+      for (const auto& level : msg->levels)
+      {
+        _data->_detector->add_level(level.name, level.elevation);
+      }
+    });
+}
+
+void HumanDetector::make_detector()
+{
+  _data->_camera_name = this->declare_parameter("camera_name", "camera1");
+  RCLCPP_INFO(
+    this->get_logger(),
+    "Setting parameter camera_name to %s", _data->_camera_name.c_str()
+  );
+  _data->_camera_image_topic = _data->_camera_name + "/image_rect";
```

Comment:
Why are you appending `image_rect`? The parameter value should be the exact topic that you should subscribe to